http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -137,9 +140,19 @@ class ClusterMonitorActor(val cluster: Cluster,
     * @return ScheduleMonitorPass
     */
   private def handleNotReadyCluster(status: ClusterStatus, instances: Set[Instance]): Future[ClusterMonitorMessage] = {
-    logger.info(s"Cluster ${cluster.projectNameString} is not ready yet (cluster status = $status, instance statuses = ${instances.groupBy(_.status).mapValues(_.size)}). Checking again in ${monitorConfig.pollPeriod.toString}.")
-    persistInstances(instances).map { _ =>
-      ScheduleMonitorPass
+    val currTimeElapsed: FiniteDuration = startupTimings.getOrElse(cluster.id, 0 seconds)
+    logger.info(s"Cluster ${cluster.projectNameString} has taken ${currTimeElapsed.toString} so far, continuing to monitor.")
+
+    if (currTimeElapsed > monitorConfig.creationTimeLimit && status == ClusterStatus.Creating) {
```

Comment:
Sorry if this is scope creep - I think we should enforce a timeout for the other "monitored" statuses too:
- Creating
- Starting
- Stopping
- Deleting
- Updating

We have definitely seen clusters stuck in these states.